packages/node/src/retryHandler.ts

```diff
@@ -1,28 +1,48 @@
 import { Event, Options, Transport, TransportOptions, Payload, Status, Response, RetryClass } from '@amplitude/types';
 import { HTTPTransport } from './transports';
-import { DEFAULT_OPTIONS, BASE_RETRY_TIMEOUT } from './constants';
-import { asyncSleep, collectInvalidEventIndices } from '@amplitude/utils';
+import { DEFAULT_OPTIONS, BASE_RETRY_TIMEOUT_DEPRECATED } from './constants';
+import { asyncSleep, collectInvalidEventIndices, logger } from '@amplitude/utils';
 
 interface RetryMetadata {
   shouldRetry: boolean;
   shouldReduceEventCount: boolean;
   eventIndicesToRemove: number[];
 }
 
+/**
+ * Converts deprecated maxRetries option to retryTimeouts
+ */
+function convertMaxRetries(maxRetries: number): number[] {
+  const retryTimeouts = [];
+  let currentTimeout = BASE_RETRY_TIMEOUT_DEPRECATED;
+  for (let i = 0; i < maxRetries; i++) {
+    retryTimeouts.push(currentTimeout);
+    currentTimeout *= 2;
+  }
+  return retryTimeouts;
+}
+
 export class RetryHandler implements RetryClass {
   protected readonly _apiKey: string;
 
   // A map of maps to event buffers for failed events
   // The first key is userId (or ''), and second is deviceId (or '')
   private readonly _idToBuffer: Map<string, Map<string, Event[]>> = new Map<string, Map<string, Event[]>>();
-  private readonly _options: Options;
+  protected readonly _options: Options;
   private readonly _transport: Transport;
   private _eventsInRetry = 0;
 
   public constructor(apiKey: string, options: Partial<Options>) {
     this._apiKey = apiKey;
     this._options = Object.assign({}, DEFAULT_OPTIONS, options);
     this._transport = this._options.transportClass ?? this._setupDefaultTransport();
+    if (this._options.maxRetries !== undefined) {
+      logger.warn(
+        'DEPRECATED: Please use retryTimeouts. It will be converted to retryTimeouts with exponential wait times (i.e. 100ms -> 200ms -> 400ms -> ...)',
```

Comment:
quick note - the retry handler is set up in `nodeClient.ts` before the logger is - this is my own oversight. the logger should be set up first so this will log correctly.

Reply:
Should we [swap these two lines then](https://github.com/amplitude/Amplitude-Node/blob/improve-sleep-retry/packages/node/src/nodeClient.ts#L27-L28)?

Reply:
yep!